core/src/avm2/function.rs

```diff
@@ -159,10 +164,103 @@ impl<'gc> Executable<'gc> {
                     subclass_object,
                     callee,
                 )?;
-
+                activation
+                    .context
+                    .avm2
+                    .push_call(activation.context.gc_context, self.clone())?;
                 activation.run_actions(bm.method)
             }
+        };
+        activation
+            .context
+            .avm2
+            .pop_call(activation.context.gc_context);
+        ret
+    }
+
+    pub fn bound_superclass(&self) -> Option<ClassObject<'gc>> {
+        match self {
+            Executable::Native(NativeExecutable {
+                bound_superclass, ..
+            }) => *bound_superclass,
+            Executable::Action(BytecodeExecutable {
+                bound_superclass, ..
+            }) => *bound_superclass,
+        }
+    }
+
+    pub fn write_full_name(&self, output: &mut WString) {
+        let class_def = self.bound_superclass().map(|superclass| {
+            let class_def = superclass.inner_class_definition();
+            let name = class_def.read().name().to_qualified_wstring();
+            output.push_str(&name);
+            class_def
+        });
+        match self {
+            Executable::Native(NativeExecutable { method, .. }) => {
+                output.push_char('/');
+                output.push_utf8(&method.name);
+            }
+            Executable::Action(BytecodeExecutable { method, .. }) => {
+                // NOTE: The name of a bytecode method refers to the name of the trait that contains the method,
+                // rather than the name of the method itself.
+                if let Some(class_def) = class_def {
+                    if Gc::ptr_eq(
+                        class_def.read().class_init().into_bytecode().unwrap(),
+                        *method,
+                    ) {
+                        output.push_utf8("$cinit");
+                    } else if !Gc::ptr_eq(
+                        class_def.read().instance_init().into_bytecode().unwrap(),
+                        *method,
+                    ) {
+                        // TODO: Ideally, the declaring trait of this executable should already be attached here, that way
+                        // we can avoid needing to lookup the trait like this.
+                        let class_def = class_def.read();
+                        let mut method_trait = None;
+                        // First search instance traits for the method
+                        for t in class_def.instance_traits() {
+                            if let Some(m) = t.as_method() {
+                                let bytecode = m.into_bytecode().unwrap();
```

Comment:
This should check if this is actually a bytecodr method - it could be a native method.

Reply:
Thanks, I missed that - I guess this behavior is different now that bytecode classes can have native methods